### PR TITLE
banner-plugin typings

### PR DESCRIPTION
**Pull request by Legends**
**What kind of change does this PR introduce?**
banner-plugin typings

**Did you add tests for your changes?**
N/A

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
nothing

### CODE REVIEW

File: lib/BannerPlugin.js [hunk 1]
```diff
@@ -12,6 +12,21 @@ const Template = require("./Template");
 const validateOptions = require("schema-utils");
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
+/**
+ * BannerPlugin Options class.
+ * @typedef {Object|string} BannerOptions
+ * @property {string} banner The banner as string, it will be wrapped in a comment
```
Comment (ooflorent) [inline lib/BannerPlugin.js]: `Function` is also valid here.

Comment (ooflorent) [reply lib/BannerPlugin.js]: More here: https://github.com/webpack/webpack/blob/master/lib/BannerPlugin.js#L35-L39

Comment (Legends) [reply lib/BannerPlugin.js]: Has the BannerPlugin been updated meanwhile??? I can see the lines on the server version... will add the change. Thanks!

Comment (Legends) [reply lib/BannerPlugin.js]: after `git pull upstream master` I do: `yarn lint`
1.) ESLint couldn't find the plugin "eslint-plugin-jest".
2.) Error: Cannot find module 'fast-deep-equal'

Comment (ooflorent) [reply lib/BannerPlugin.js]: Okay, let's try to debug this. First, be sure to have to correct upsteam:

```
% git remote -v
origin		git@github.com:Legends/webpack.git (fetch)
origin		git@github.com:Legends/webpack.git (push)
upstream	git@github.com:webpack/webpack.git (fetch)
upstream	git@github.com:webpack/webpack.git (push)
```

Then update your `upsteam` remote:

```
% git fetch upstream
```

You can now rebase your branch:

```
% git rebase upstream/master
```

You could also reset your `master` to `upstream/master` to keep your branches in sync:

```
% git checkout master
% git reset --hard upstream/master
```

Once your branch is cleaned up, running `yarn` should fix your dependency issues.

Comment (Legends) [reply lib/BannerPlugin.js]: Thanks a lot @ooflorent, no dep issue anymore, but two lint errors in BannerPlugin.js:
"webpack-sources" and "schema-utils" is not found (node/no-missing-require) line 8 + 12

Comment (Legends) [reply lib/BannerPlugin.js]: Restarting VsCode helps....

Comment (Legends) [reply lib/BannerPlugin.js]: After "Ran all test suites" I also get this red colored message, duno if it's important:

Force exiting Jest
Have you considered using `--detectOpenHandles` to detect async operations that kept running after all tests finished?

File: lib/BannerPlugin.js [hunk 2]
```diff
@@ -12,6 +12,21 @@ const Template = require("./Template");
 const validateOptions = require("schema-utils");
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
+/**
+ * BannerPlugin Options class.
+ * @typedef {Object|string|Function} BannerOptions
+ * @property {string} banner The banner as string, it will be wrapped in a comment
```
Comment (ooflorent) [inline lib/BannerPlugin.js]: Could also accept a function here

Comment (Legends) [reply lib/BannerPlugin.js]: I've added `Function` to BannerOptions --> `@typedef {......|......|Function}`  .
In your opinion where do I have to add `Function`?

Comment (Legends) [reply lib/BannerPlugin.js]: Not sure about this, but IMO I would define `BannerOptions` like ` @typedef {Object} BannerOptions` with all its properties and then, I would define the constructor param like this:
` @param {BannerOptions|string|Function} options`, but if I do it like this, I get a bunch of TS errors..

Comment (Legends) [reply lib/BannerPlugin.js]: `options.banner` ...done!

**Comment by Legends** [comment]
Closing this PR as I have deleted this branch, opening new [PR](https://github.com/webpack/webpack/pull/7568) instead

**Pull request closed without merging**